core/arch/arm/include/kernel/spinlock.h

```diff
@@ -69,7 +69,7 @@ static inline unsigned int cpu_spin_trylock(unsigned int *lock)
 
 	assert(thread_irq_disabled());
 	locked = __cpu_spin_trylock(lock);
-	if (locked)
+	if (!locked)
```

Comment:
oohh! my mistake, you're right :) `x0` remains to the lock address.
I'll propose a new change, overriding my current proposal.